src/appleseed.shaders/include/appleseed/fractal/as_fractal_helpers.h

```diff
@@ -34,25 +34,33 @@
 #define AS_FRACTAL_HELPERS_H
 
 #include "appleseed/fractal/as_noise_helpers.h"
-#include "appleseed/math/as_math_helpers.h"
 
 float fBm(
     point surface_point,
+    float initial_time,
     float filter_width,
     float amplitude,
     int octaves,
     float lacunarity,
     float gain)
 {
-    point pp = surface_point;
-    float amp = amplitude, fw = filter_width, sum = 0.0;
+    point xyz = surface_point;
+    float amp = amplitude, filter_size = filter_width;
+    float current_time = initial_time, sum = 0.0;
 
     for (int i = 0; i < octaves; ++i)
     {
-        sum += amp * filtered_snoise(pp, fw);
+        if (!amp)
```

Comment:
@FrancoisBeaune not really, because *amp* is a copy of *amplitude* that is modified within the loop by *ratio* after accumulating the iteration's noise. It's entirely possible to have a ratio of 0, which would result on the fractal noise returning only the first frequency result.

![ratio](https://cloud.githubusercontent.com/assets/387352/22812103/beea4552-ef7d-11e6-8ba2-68838ac5c5e3.jpg)

As far as potential optimization goes we can either test *amp* within the loop. 
Or test for *ratio* outside the loop, if 0, run the sequence/call for the base frequency only, if > 0, proceed to accumulate all the frequencies. I'm not sure we would gain anything with this 2nd approach but i haven't profiled this.
 
As a side note, the global parameter *amplitude* should be tested as well (we're not testing it yet) but it's way before we reach any of these calls. I'm leaving this global amplitude test for later since i want to find common ground with the remaining noise/fractal nodes.